tests/phpunit/api/v3/ContributionTest.php

```diff
@@ -2668,6 +2668,148 @@ public function testRepeatTransactionPassedInFinancialType(): void {
+  }
+
+  /**
+   * Test contribution_page_id with a recurring contribution.
+   *
+   * The contribution_page_id is stored at the recurring contribution
+   * at the first contribution and at the template contribution.
+   *
+   * This test checks whether the contribution_page_id is taken from the latest contribution.
+   * We change the contribution_page_id at the second contribution and then
+   * we repeat the transaction and check the contribution_page_id again.
+   */
+  public function testRepeatTransactionWithChangedContributionPageId(): void {
+    $contributionPage = $this->contributionPageCreate(['receipt_from_name' => 'CiviCRM LLC', 'receipt_from_email' => 'contributionpage@civicrm.org', 'is_email_receipt' => 1]);
+    $contributionPage2 = $this->contributionPageCreate(['receipt_from_name' => 'CiviCRM LLC', 'receipt_from_email' => 'contributionpage@civicrm.org', 'is_email_receipt' => 1]);
+    $contributionRecurParams = [
+      'contribution_page_id' => $contributionPage['id'],
+    ];
+    $contributionParams = [
+      'contribution_page_id' => $contributionPage['id'],
+      'trxn_id' => 'number_1',
+    ];
+    $originalContribution = $this->setUpRepeatTransaction($contributionRecurParams, 'single', $contributionParams);
+
+    $this->callAPISuccess('Contribution', 'repeattransaction', [
+      'contribution_recur_id' => $originalContribution['contribution_recur_id'],
+      'contribution_status_id' => 'Completed',
+      'trxn_id' => 'number_2',
+    ]);
+
+    $contribution = Contribution::get()
+      ->addWhere('trxn_id', '=', 'number_2')
+      ->setSelect(['id', 'contribution_page_id'])
+      ->execute()
+      ->single();
+    $this->assertEquals($contributionPage['id'], $contribution['contribution_page_id']);
+
+    Contribution::update()
+      ->setValues(['contribution_page_id' => $contributionPage2['id']])
+      ->addWhere('id', '=', $contribution['id'])
+      ->execute();
+
+    $this->callAPISuccess('contribution', 'repeattransaction', [
+      'contribution_recur_id' => $originalContribution['contribution_recur_id'],
+      'contribution_status_id' => 'Completed',
+      'trxn_id' => 'number_3',
+    ]);
+
+    $contribution = Contribution::get()
+      ->addWhere('trxn_id', '=', 'number_3')
+      ->setSelect(['id', 'contribution_page_id'])
+      ->execute()
+      ->single();
+    $this->assertEquals($contributionPage2['id'], $contribution['contribution_page_id']);
+  }
+
+  /**
+   * Test contribution_page_id with a recurring contribution.
+   *
+   * The contribution_page_id is stored at the recurring contribution
+   * at the first contribution and at the template contribution.
+   *
+   * This test checks whether the contribution_page_id is taken from the template contribution.
+   * We change the contribution_page_id at the template contribution and then
+   * we repeat the transaction and check the contribution_page_id again.
```

Comment:
Just reading this - to be in line with the other fields the contribution_recur contribution page ID should take precedence, if different

Reply:
This is updated